Lib9c/Action/Garages/UnloadFromMyGarages.cs

```diff
@@ -0,0 +1,196 @@
+            IEnumerable<(HashDigest<SHA256> fungibleId, int count)>? fungibleIdAndCounts)
+        {
+            (
+                FungibleAssetValues,
+                InventoryAddr,
+                FungibleIdAndCounts
+            ) = GarageUtils.MergeAndSort(
+                fungibleAssetValues,
+                inventoryAddr,
+                fungibleIdAndCounts);
+        }
+
+        public UnloadFromMyGarages()
+        {
+        }
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            (
+                FungibleAssetValues,
+                InventoryAddr,
+                FungibleIdAndCounts
+            ) = GarageUtils.Deserialize(plainValue["l"]);
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            if (context.Rehearsal)
+            {
+                return states;
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context);
+            ValidateFields(addressesHex);
+            states = TransferFungibleAssetValues(
+                context.Signer,
+                states);
+            return TransferFungibleItems(
+                context.Signer,
+                states);
+        }
+
+        private void ValidateFields(string addressesHex)
+        {
+            if (FungibleAssetValues is null &&
+                FungibleIdAndCounts is null)
+            {
+                throw new InvalidActionFieldException(
+                    $"[{addressesHex}] Either FungibleAssetValues or FungibleIdAndCounts " +
+                    "must be set.");
+            }
+
+            if (FungibleAssetValues != null)
+            {
+                foreach (var (_, value) in FungibleAssetValues)
+                {
+                    if (value.Sign < 0)
+                    {
+                        throw new InvalidActionFieldException(
+                            $"[{addressesHex}] FungibleAssetValue.Sign must be positive.");
+                    }
+                }
+            }
+
+            if (FungibleIdAndCounts is null)
+            {
+                return;
+            }
+
+            if (!InventoryAddr.HasValue)
+            {
+                throw new InvalidActionFieldException(
+                    $"[{addressesHex}] {nameof(InventoryAddr)} is required when " +
+                    $"{nameof(FungibleIdAndCounts)} is set.");
+            }
+
+            foreach (var (fungibleId, count) in FungibleIdAndCounts)
+            {
+                if (count < 0)
+                {
+                    throw new InvalidActionFieldException(
+                        $"[{addressesHex}] Count of fungible id must be positive." +
+                        $" {fungibleId}, {count}");
+                }
+            }
+        }
+
+        private IAccountStateDelta TransferFungibleAssetValues(
```

Comment:
Do we have to write down this function in load/unload separately? (same as TransferFungibleItems)
In my short sight, just switching sender and recipient do the thing right.
If I'm missing something, feel free to let me know.

Reply:
You are right. Thanks!